libutils/string_lib.c

```diff
@@ -528,6 +528,27 @@ int StringToLong(const char *str, long *value_out)
     return 0;
 }
 
+/**
+ * @brief Convert a string representing a real number to int, discarding the decimal part.
+ *
+ * @param[in] str String matching the regex \d+\.\d+, cannot be NULL
```

Comment:
If the caller of this function has to assert this first, they don't need this function. It needs to be able to work with at least just `\d+` strings.

Reply:
Actually the regex is off. It should be `\d+(\.\d+)?`. Because if only an integer is passed then it still works. The test i have on the other pr confirms it. In essence it scans the string to find the integer part but if there _only_ an integer then it still works since the entire string _is_ the integer part.